src/program_generator/common.py

```diff
@@ -0,0 +1,101 @@
+import csv
+from math import ceil
+
+import constants
+
+"""
+Common tools for program generation. To be organized if needed, for now just bag of functions.
+"""
+    
+
+def generate_single_marginal(single_op_pushes, operation, op_count):
```

Comment:
The code here has been distilled out of `pg_marginal` and `pg_arguments`, but this isn't a refactor - there have been some changes to the logic